src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -481,7 +481,9 @@ std::unique_ptr<EventEngine::DNSResolver> PosixEventEngine::GetDNSResolver(
   grpc_core::Crash("unimplemented");
 }
 
-bool PosixEventEngine::IsWorkerThread() { grpc_core::Crash("unimplemented"); }
+bool PosixEventEngine::IsWorkerThread() {
```

Comment:
My read here is that this returns true if the thread is a worker thread is from *any* event engine - if that's the case we should rephrase the API to make this not a member function - a member function implies "this is a worker thread from *this* event engine instance"

Reply:
That's a good point. Given that we have a solution that does not require `IsWorkerThread`, I will back out this implementation and leave decisions about the public API for another day. I imagine the "any EventEngine thread" version will be more useful, but we'll get a better idea of that in the next few months.